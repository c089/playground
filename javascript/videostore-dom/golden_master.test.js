const jsdom = require('jsdom');
const fs = require('fs');
const source = fs.readFileSync('./index.js');

const buildFixture = (customerName, rentals) => {
    const html = `
            <!DOCTYPE html>
            <html>
            <head>
            <meta charset="utf-8">
            <title>JS Bin</title>
            <style>span.amount { float: right; }</style>
            </head>
            <body>
                <div id="rentals"></div>
                <script type="text/javascript">
                    ${source}
                    showStatement(${JSON.stringify(customerName)}, ${JSON.stringify(rentals)});
                </script>
            </body>
            </html>`;

    return new jsdom.JSDOM(html, { runScripts: "dangerously" });
};

test('verify', () => {
    const customerName = 'foo';
    const rentals = [
        {daysRented: 2, movie: { title: 'movie 1', priceCode: 'REGULAR' }},
        {daysRented: 1, movie: { title: 'other', priceCode: 'CHILDRENS'} },
        {daysRented: 5, movie: { title: 'nonsense', priceCode: 'NEW_RELEASE'}}];

    const dom = buildFixture(customerName, rentals);

    expect(dom.window.document.getElementById('rentals')).toMatchSnapshot();
});
