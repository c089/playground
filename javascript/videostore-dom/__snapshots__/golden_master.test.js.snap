// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verify 1`] = `
<div
  id="rentals"
  style="width: 500px;"
>
  Rental Record for: foo

  <div>
    <span>
      movie 1
    </span>
    <span
      class="amount"
    >
      2 EUR
    </span>
  </div>
  <div>
    <span>
      other
    </span>
    <span
      class="amount"
    >
      1.5 EUR
    </span>
  </div>
  <div>
    <span>
      nonsense
    </span>
    <span
      class="amount"
    >
      15 EUR
    </span>
  </div>
  <div>
    Amount owed is 18.5 EUR
  </div>
  <div>
    You earned 4 frequent renter points
  </div>
</div>
`;
